### PR TITLE
Gender implementation for employees

### PR DESCRIPTION
**Pull request by JoaFS**

### CODE REVIEW

File: src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java [hunk 1]
```diff
@@ -5,22 +5,26 @@
 import java.util.List;
 import java.util.UUID;
 import java.util.stream.Collectors;
+
+import com.entropyteam.entropay.employees.models.*;
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java:9]: please revert this change. It is not a good practice to use `*` in the imports.

File: src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java [hunk 2]
```diff
@@ -5,22 +5,26 @@
...
+
 public record EmployeeDto(UUID id,
                           @NotNull(message = "Internal ID is mandatory")
                           String internalId,
                           @NotNull(message = "First Name is mandatory")
                           String firstName,
                           @NotNull(message = "Last Name is mandatory")
                           String lastName,
+                          @NotNull(message = "Gender is mandatory")
+//                          String Gender,
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java:26]: please remove commented code

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java [hunk 3]
```diff
@@ -3,14 +3,8 @@
...
-import jakarta.persistence.FetchType;
-import jakarta.persistence.JoinColumn;
-import jakarta.persistence.JoinTable;
-import jakarta.persistence.Table;
-import jakarta.persistence.Entity;
-import jakarta.persistence.ManyToMany;
-import jakarta.persistence.OneToMany;
-import jakarta.persistence.ManyToOne;
+
+import jakarta.persistence.*;
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/models/Employee.java:7]: please fix the imports

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java [hunk 4]
```diff
@@ -23,6 +17,9 @@ public class Employee extends BaseEntity {
     private String internalId;
     private String firstName;
     private String lastName;
+    //private String Gender;
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/models/Employee.java:20]: please remove commented code

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java [hunk 5]
```diff
@@ -104,6 +102,14 @@ public void setLastName(String lastName) {
         this.lastName = lastName;
     }
 
+    public Gender getGender() { return gender;  }
+
+    public void setGender(Gender gender) {  this.gender = gender;   }
+
+    //    public String getGender() { return Gender;  }
+//
+//    public void setGender(String Gender) { this.Gender = Gender;    }
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/models/Employee.java:111]: please remove commented code

File: src/main/java/com/entropyteam/entropay/employees/models/Gender.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.entropyteam.entropay.employees.models;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+@JsonFormat(shape = JsonFormat.Shape.STRING)
+public enum Gender{Male, Female, NO_BINARY}
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/models/Gender.java:6]: please change to MALE and FEMALE to keep the format convention

File: src/main/resources/db/migration/V42.0__DDL_update_employee_table.sql [hunk 7]
```diff
@@ -44,3 +44,4 @@ UPDATE employee SET country_id = country.id FROM country WHERE LOWER(employee.co
 
 ALTER TABLE employee ALTER COLUMN country_id SET NOT NULL;
 ALTER TABLE employee DROP COLUMN country;
+ALTER TABLE employee ADD COLUMN gender varchar(10);
```
Comment (eazaretzky) [inline src/main/resources/db/migration/V42.0__DDL_update_employee_table.sql:47]: migrations are not supposed to change. Please revert this change and create a new one

**Review by eazaretzky — CHANGES_REQUESTED** [review]

File: src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java (hunk 1 above)
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java]: Reverted

File: src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java (hunk 2 above)
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/dtos/EmployeeDto.java]: Removed

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java (hunk 3 above)
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/models/Employee.java]: Fixed

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java (hunk 4 above)
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/models/Employee.java]: Removed

File: src/main/java/com/entropyteam/entropay/employees/models/Employee.java (hunk 5 above)
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/models/Employee.java]: Removed

File: src/main/java/com/entropyteam/entropay/employees/models/Gender.java (hunk 6 above)
Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/models/Gender.java]: Changed

File: src/main/resources/db/migration/V42.0__DDL_update_employee_table.sql (hunk 7 above)
Comment (JoaFS) [reply src/main/resources/db/migration/V42.0__DDL_update_employee_table.sql]: Reverted. New migration file created

File: src/main/java/com/entropyteam/entropay/employees/models/Gender.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package com.entropyteam.entropay.employees.models;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+@JsonFormat(shape = JsonFormat.Shape.STRING)
+public enum Gender{MALE, FEMALE, NO_BINARY}
```
Comment (eazaretzky) [inline src/main/java/com/entropyteam/entropay/employees/models/Gender.java:6]: one last and small change, it should be NON_BINARY right?

Comment (JoaFS) [reply src/main/java/com/entropyteam/entropay/employees/models/Gender.java]: Had exactly the same doubt last week but in the Basecamp ticket said "Male, Female, No Binary" and I took that for reference to create the enum. Anyway, I changed it and now it's NON_BINARY

**Review by agustinb4 — APPROVED** [review]

**Review by eazaretzky — APPROVED** [review]
